tests/test_resolver.py

```diff
@@ -575,3 +587,74 @@ def resolve(self, *args, **kwargs):
             resolver=FakePipResolver(),
             compatible_existing_constraints={},
         )
+
+
+@pytest.mark.parametrize(
+    ("constraints", "conflicting_existing", "compatible_existing"),
+    (
+        (
+            [
+                "poetry==1.6.1",
+            ],
+            ["cachecontrol[filecache]==0.12.14"],
+            ["doesnotexist==1.0.0"],
+        ),
+        (
+            [
+                "poetry==1.6.1",
+            ],
+            ["keyring==22.1.0", "pkginfo==1.7.2"],
+            ["platformdirs==3.0.0", "doesnotexist==1.0.0"],
+        ),
+    ),
+)
+def test_backtracking_resolver_drops_existing_conflicting_constraints(
+    backtracking_resolver: Callable[..., BacktrackingResolver],
+    from_line: Callable[..., InstallRequirement],
+    constraints: Sequence[str],
+    conflicting_existing: Sequence[str],
+    compatible_existing: Sequence[str],
+) -> None:
+    def wrap_resolution_impossible(*args, **kwargs):
+        """
+        Raise a ``DistributionNotFound`` exception that has a ``ResolutionImpossible``
+        exception as its cause.
+        """
+        try:
```

Comment:
Since you raise `ResolutionImpossible` anyway, I'd drop this `try` case.

Reply:
It took me a moment but I think I now understood what you meant. 😄 Changed as well.